rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/FileMetadata.java

```diff
@@ -41,30 +48,68 @@
   private long size;
   private long lastModified;
   private long currentFileSize;
-
   private boolean needsRefresh = true;
+  private int generation;
 
   int bitmapFileSizeBytes;
   ByteBufferBitmap blockBitmap;
 
   static Striped<Lock> stripes = Striped.lock(20000);
+  //  Maintains generation number for remote file
+  //  If multiple threads call get of guava cache , then all are blocked and computation is done for one
+  //  and values is returned to other blocked threads.
+  static Cache<String, Integer> generationNumber = CacheBuilder.newBuilder()
```

Comment:
These are initialized in BookKeeper. Do this:
1. Move these declarations into BookKeeper
2. Drop static
3. Drop default definitions
4. Make BookKeeper.initializeCache as non-static so that you can define these non static objects in it.